migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker/unexecutable_step_check.rs

```diff
@@ -50,6 +53,27 @@ impl Check for UnexecutableStepCheck {
 
                 Some(message)
             }
+            UnexecutableStepCheck::AddedRequiredFieldToTableWithPrismaLevelDefault { table, column } => {
+                let message = |details| {
+                    format!(
+                        "The required column `{column}` was added to the `{table}` table with a prisma-level default value. {details}. Please create add this column as optional, then populate it before making it required.",
+                        table = table,
+                        column = column,
+                        details = details,
+                    )
+                };
+
+                let message = match database_checks.get_row_count(table) {
+                    Some(0) => return None, // Adding a required column is possible if there is no data
+                    Some(row_count) => message(format_args!(
```

Comment:
Just a general question for later: Would an existence check not be much faster. And we do not really care about the number of rows anyways right?

Reply:
Indeed, that would make a lot of sense.